ember-client/native/src/main/scala/org/http4s/ember/client/EmberClientBuilderCompanionPlatform.scala

```diff
@@ -17,14 +17,21 @@
 package org.http4s.ember.client
 
 import cats.effect.Async
+import cats.effect.IO
 import fs2.io.net.unixsocket.UnixSockets
 import org.typelevel.log4cats.Logger
 import org.typelevel.log4cats.noop.NoOpLogger
 
 private[client] trait EmberClientBuilderPlatform {
 
-  private[client] def defaultUnixSockets[F[_]]: Option[UnixSockets[F]] =
-    None
+  private[client] def defaultUnixSockets[F[_]](implicit F: Async[F]): Option[UnixSockets[F]] =
+    if (F eq IO.asyncForIO)
+      Some(UnixSockets.forLiftIO[IO].asInstanceOf[UnixSockets[F]])
+    else
+      None
+
+  @deprecated("bincompat stub", "0.23.31")
+  def defaultUnixSockets(): Option[Nothing] = None
```

Comment:
It seems like you then must call it with `()` too...
```
[error] -- [E100] Syntax Error: /home/runner/work/http4s/http4s/ember-client/shared/src/main/scala/org/http4s/ember/client/EmberClientBuilder.scala:347:22 
[error] 347 |              .orElse(defaultUnixSockets)
[error]     |                      ^^^^^^^^^^^^^^^^^^
[error]     |method defaultUnixSockets in trait EmberClientBuilderPlatform must be called with () argument
[error]     |
[error]     | longer explanation available when compiling with `-explain`
[error] one error found
```
https://github.com/http4s/http4s/actions/runs/12552012104/job/34997348764?pr=7161#step:15:44

Reply:
Actually, the goal is to avoid calling it at all. Hmm ...